tests/src/Interop/ArrayMarshalling/SafeArray/SafeArrayNative.cpp

```diff
@@ -0,0 +1,313 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include <xplatform.h>
+#include <oleauto.h>
+#include <algorithm>
+#include "platformdefines.h"
+#include "Helpers.h"
+
+extern "C" DLL_EXPORT BOOL STDMETHODCALLTYPE XorBoolArray(SAFEARRAY* d, BOOL* result)
```

Comment:
We are getting to the point where this model of testing is more burden than benefit.

Instead of returning a `BOOL`, just return the `HRESULT`. Tests should almost always just throw an exception or return an error code. The `BOOL` concept is a relic that should be purged from the codebase. `HRESULT`s will be properly converted to an `Exception` and in most cases translated. This will also allow you to follow the standard pattern of `RETURN_IF_FAILED()`.